### PR TITLE
feat: #55 add table with favourite ways to user page

### PR DESCRIPTION
**Pull request by Howl404**

### CODE REVIEW

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts [hunk 1]
```diff
@@ -79,17 +79,20 @@ export class WayPreviewDAL {
   /**
    * Get User ways preview based of provided type
    */
-  public static async getUserWaysPreview(uuid: string, type: "OwnWays" | "MentoringWays"): Promise<WayPreview[]> {
+  public static async getUserWaysPreview(uuid: string, type: "Own" | "Mentoring" | "Favourite"): Promise<WayPreview[]> {
```
Comment (SergioAJS) [inline mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:82]: "Favorite" is the right word. Please check the correct spelling of the word in all files

**Review by SergioAJS — CHANGES_REQUESTED** [review]

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts (hunk 1 above)
Comment (Howl404) [reply mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:82]: [Favorite and Favourite are both correct](https://www.merriam-webster.com/grammar/favorite-or-favourite-usage)

**Review by SergioAJS — APPROVED** [review]

File: mw-webapp/src/service/WayService.ts [hunk 2]
```diff
@@ -47,12 +48,25 @@ export class WayService {
...
     return mentoringWays;
   }
 
+  /**
+   * Get WaysDTO of user favourite ways
+   */
+  public static async getFavouriteWaysDTO(uuid: string): Promise<WayDTO[]> {
+    const userDTO = await UserService.getUserDTO(uuid);
+    const waysRef = collection(db, PATH_TO_WAYS_COLLECTION);
+    const favouriteWaysQuery = query(waysRef, where("uuid", "in", userDTO.favoriteWayUuids));
```
Comment (SergioAJS) [inline mw-webapp/src/service/WayService.ts:65]: On line 65, there are two spellings of the word "favorite". let's choose one variant

Comment (Howl404) [reply mw-webapp/src/service/WayService.ts]: Changed to favorite everywhere

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/logic/waysTable/FavoriteWaysTable.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Favorite ways table props
+ */
+interface FavoriteWaysTableProps {
+
+  /**
+   * User Uuid
+   */
+  uuid: string;
+}
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/waysTable/FavoriteWaysTable.tsx:16]: Each table has the same interface with only one prop uuid/ What do you think about create one common props for all waysTables? Is it good or bad idea?

Comment (Ekaterina1994) [inline mw-webapp/src/logic/waysTable/FavoriteWaysTable.tsx:16]: Each table has the same interface with only one prop uuid/ What do you think about create one common props for all waysTables? Is it good or bad idea?

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/logic/waysTable/FavoriteWaysTable.tsx (hunk 3 above)
Comment (tritonJS826) [reply mw-webapp/src/logic/waysTable/FavoriteWaysTable.tsx:16]: if it is common you can even create a generic like WithChildrenProp, but for your purposes (for adding uuid)

Comment (Howl404) [reply mw-webapp/src/logic/waysTable/FavoriteWaysTable.tsx:16]: Maybe I'm not understanding something

Interface PropsWithUuid {
  uuid: string
}

Comment (Ekaterina1994) [reply mw-webapp/src/logic/waysTable/FavoriteWaysTable.tsx:16]: If speak about generics then maybe you can use :
type PropsWithUuid<T> = {
  uuid: T;
};

But with string it's more clear/ Don't know

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

**Review by Ekaterina1994 — APPROVED** [review]

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**